File: frontend/src/api/dto/verifyOtpRequest.ts
/**
 * OTP verification request
 */
export interface VerifyOtpRequest {
  /**
   * Email address
   * @maxLength 250
   * @pattern \S
   */
  email: string
  /**
   * 6-digit OTP code
   * @pattern ^\d{6}$
   */
  code: string
}

File: backend/src/main/java/fr/pedalons/dto/auth/request/VerifyOtpRequest.java
package fr.pedalons.dto.auth.request;

import fr.pedalons.dto.validation.ValidateSchema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Builder;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "OTP verification request")
@ValidateSchema
@Builder
public record VerifyOtpRequest(
    @NotBlank
        @Email
        @Size(max = 250)
        @Schema(description = "Email address", examples = "user@example.com")
        String email,
    @NotBlank
        @Pattern(regexp = "^\\d{6}$", message = "Code must be 6 digits")
        @Schema(description = "6-digit OTP code", examples = "123456")
        String code) {}
